src/markdown/doc_of_md.ml

```diff
@@ -64,11 +64,15 @@ let textloc_to_loc ~locator textloc =
   let point_of_line_and_byte_pos ~locator:(location, line_trim_counts) l pos =
     let line_num, line_pos = l in
     let line = location.Lexing.pos_lnum + line_num - 1 in
-    let column = line_trim_counts.(line_num - 1) + (pos - line_pos) in
-    let column =
-      match line_num with 1 -> comment_col ~location + column | _ -> column
-    in
-    { Loc.line; column }
+    try
+      let column = line_trim_counts.(line_num - 1) + (pos - line_pos) in
+      let column =
+        match line_num with 1 -> comment_col ~location + column | _ -> column
+      in
+      { Loc.line; column }
+    with _ ->
+      (* Presumably this is the above-mentioned bug that's being hit. *)
+      { Loc.line = -1; column = -1 }
```

Comment:
The only thing that raises a `Invalid_argument` is the array access at line 68. I think this could be handled better. It's also possible that `locator` is not computed right by `massage_comment` as it's so complicated.

Reply:
There's quite a lot we can do with the markdown code, but I don't want to spend too much time on it until I'm sure people want it and are using it.